From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Tue, 14 Apr 2026 15:12:55 +0100
Subject: test concurrency

---
 tests/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index d124ac2..4125783 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -27,5 +27,6 @@ EXTRA_DIST = docheck1.sh docheck2.sh docheck3.sh emfs/check1.emf emfs/check3.emf
 
 MOSTLYCLEANFILES = check1.emf check2.emf check3.emf
 
-# Dependency of output from docheck1 for docheck2
+# check2 and check3 read check1.emf produced by check1
 docheck2.log: docheck1.log
+docheck3.log: docheck1.log
